server/api/apps/quiz_create/views.py

```diff
@@ -0,0 +1,171 @@
+from rest_framework.decorators import api_view
+from django.http import HttpResponse, JsonResponse
+from django.shortcuts import get_object_or_404
+from rest_framework import status
+from .serializers import QuestionInfoSerializer
+from api.apps.shared_models.models import quiz_models
+from api.apps.shared_models.serializers import quiz_serializers
+
+
+# this method associates a tag with a question or adds a new tag to the
+# database and returns the Tag object
+def AddTagToQuestion(question, tag_name):
+    try:
+        tag = quiz_models.Tag.objects.get(name=tag_name)
+        question.tag_set.add(tag)
+    except quiz_models.Tag.DoesNotExist:

```

Comment:
Users shouldn't be able to create tags/topics/subjects.
Only admins should be able to do that via the admin dashboard.